pkg/bgpv1/manager/reconciler/service.go

```diff
@@ -324,6 +319,11 @@ func (r *ServiceReconciler) svcDesiredRoutes(newc *v2alpha1api.CiliumBGPVirtualR
 				desiredRoutes = append(desiredRoutes, netip.PrefixFrom(addr, addr.BitLen()))
 			}
 		case v2alpha1api.BGPClusterIPAddr:
+			// Ignore internalTrafficPolicy == Local && no local endpoints.
```

Comment:
I think ExternalTrafficPolicy checks should be moved into LoadBalancer or ExternalIP advertisments. Like InternalTrafficPolicy check here. 

Also, consider moving each advertisement types into separate methods, this switch case is getting rather big and each having different logic.

Reply:
> I think ExternalTrafficPolicy checks should be moved into LoadBalancer or ExternalIP advertisments. Like InternalTrafficPolicy check here.


does a clusterIP service not need to check ExternalTrafficPolicy? @harsimran-pabla

Reply:
I discussed this with @rastislavs, we should be clear in documentation and consistent with how `iTP` and `eTP` configurations are consumed by BGP CP. 

Ideally, these fields are independent. But since we are advertising ClusterIP to outside world it causes some confusion regarding meaning of external/internal.

This is what I propose : 
- With ClusterIP : Only `iTP` configuration is considered for advertisement, `eTP` will be ignored. 
- ExternalIP, LBIP : Only `eTP` configuration is considered for advertisement, `iTP` will be ignored.